### ISSUE TITLE
When using bash (git or cygwin) in Windows, it should be treated as bash, not cmd

### ISSUE THREAD

**Issue by matey-jack** [ImmobilienScout24/afp-cli#26]

**Comment by esc** [comment]
Can you expand on this a little, not sure exactly what you mean?

**Comment by mriehl** [comment]
I think the problem is cli looks at `os.name` to detect the suitable export format instead of being shell specific.

Same for spawning a subshell.

**Comment by matey-jack** [comment]
yes, exactly. I looked for ways to detect if bash is there, but windows doesn't have the SHELL env variable (or at least with gitbash it's not there) and no other env variable gives a clue. 
I have thought about searching PATH for bash, but not looked into the details and drawbacks of this approach.

Currently our usecase disappeared, which is why I am not working on it any more.

**Comment by esc** [comment]
As a workaround, we could define a config-file parameter to force the export format regardless of os.name .. Not perfect, but would put the user in control.

On 2 December 2015 18:02:36 CET, Robert Jack Will notifications@github.com wrote:

> yes, exactly. I looked for ways to detect if bash is there, but windows
> doesn't have the SHELL env variable (or at least with gitbash it's not
> there) and no other env variable gives a clue. 
> I have thought about searching PATH for bash, but not looked into the
> details and drawbacks of this approach.
> 
> Currently our usecase disappeared, which is why I am not working on it
> any more. 
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/ImmobilienScout24/afp-cli/issues/26#issuecomment-161365159

## 

Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by sdomme** [comment]
Maybe there is a way to get it from the hidden variable [CMDCMDLINE](http://environmentvariables.org/CmdCmdLine).

```
echo %CmdCmdLine%
"C:\Windows\system32\cmd.exe"
```

I'm done with Windows so can't test it.

**Comment by esc** [comment]
there should be an override switch (and config-file option), like `-f, --force-os` that allows you to disregard any automatic detection of the operating system. Should be trivial enough to implement.

**Comment by schlomo** [comment]
If the user wants to set the shell (bash, zsh, CMD.EXE ...) then why not give a parameter and config file item for just that? In the end the shell type determines how we have to call it and inject the variables.
